### PR TITLE
matplotlib: 3.2.1 -> 3.3.0

### PR DESCRIPTION
**Pull request by K900**
###### Motivation for this change
New upstream version.

###### Things done
Updated version, added new setup.cfg setting to build with system Freetype instead of trying to download and vendor it at build time (what even).

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FRidh** [comment]
@GrahamcOfBorg build python3.pkgs.matplotlib

**Comment by veprbl** [comment]
@grahamcofborg python2.pkgs.matplotlib

**Comment by veprbl** [comment]
@GrahamcOfBorg build python2.pkgs.matplotlib

**Comment by FRidh** [comment]
Python 2 uses a different expression.

**Comment by veprbl** [comment]
And the same setup.cfg

**Comment by K900** [comment]
Rebased on top of latest master, added the missing Pillow dependency as well. Seems like one of the checks got stuck from the previous run, so I didn't get a notification :(

**Comment by K900** [comment]
I'm not too sure what happened here with the labels and the CI failures. Can someone please help troubleshoot?

File: pkgs/development/python-modules/matplotlib/default.nix [hunk 1]
```diff
@@ -39,7 +39,7 @@ buildPythonPackage rec {
 
   propagatedBuildInputs =
     [ cycler dateutil numpy pyparsing tornado freetype kiwisolver
-      libpng mock pytz ]
+      libpng mock pytz pillow ]
```
Comment (jonringer) [inline pkgs/development/python-modules/matplotlib/default.nix:42]: pillow needs to be added to the top of the file.

It's currently "undefined"

Comment (K900) [reply pkgs/development/python-modules/matplotlib/default.nix:42]: Fixed, sorry. I've caught some sort of a cold, so brain isn't working right half the time :(

**Comment by veprbl** [comment]
This has landed as 8e5e7e2ad82a9432970fca6d066c5ce11bb19b8a

**Pull request closed without merging**